design/ceph/multus-network.md

```diff
@@ -90,111 +90,106 @@ interface will be added to the host network namespace for all nodes that will ru
-cluster, even when the CSI driver is restarted (or updated).
-
-The mover must:
-- be a privileged container
-- have `SYS_ADMIN` and `NET_ADMIN` capabilities
-- be on the host network namespace
-- have access to the `/var/run/netns` directory
-
-In order to not leave moved interfaces dangling in the host's network namespace, mover pods must
-move interfaces back to their original namespace when CSI is being terminated. The most
-straightforward way to accomplish this is to move interfaces back when the mover is being
-terminated. If an interface is moved without user applications also being removed, this will cause
-I/O disruption. Therefore, the DaemonSet should also use the `OnDelete` update strategy so that the
-pods can be updated on node reboots (or node drains).
-
-In order to better handle unexpected corner cases that leave moved interfaces in the host network
-namespace (e.g., a mover is killed abruptly rather than gracefully terminated), instead treat "move"
-operations as a disable-and-copy operation. To do this, disable the interface in the holder pod's
-network namespace, and create a copy of the interface in the host namespace with the same MAC
-address and IP config. From the user standpoint, the interface is still "moved" because the original
-is disabled, so we keep the "mover" terminology. This merely helps Rook ensure that it is not
-accidentally losing the original information.
-
-A previous iteration of this design specified the mover application as a sidecar to Ceph CSI plugin
-pods; however, this design would mean that the mover would need to be deleted and re-created
-whenever the CSI plugin is updated, possibly resulting in I/O hangs in user pods during the update.
-Keeping the mover independent allows CSI plugin updates to happen freely without complex
-interactions between it and the mover.
+The new CSI-plugin DaemonSet only contains a single container called `csi-plugin`, the container
+responsible for mounting filesystem and attaching block devices. Additionally, it has a small init
+container that removes the `/usr/local/bin/cephcsi` binary from the image. Since the main container
+will put the desired binary back in place inside its local filesystem.
+Also, the `command` to execute from the main container will change to use a new binary that will be responsible for:
+
+* watching for changes in the Kubelet HostPath volume directory about the `cephcsi` binary,
+  typically `/var/lib/kubelet/plugins/rook-ceph.<flavor>.csi.ceph.com/`, where `flavor` is either `rbd` or `cephfs`
+* if a new binary is found, it will be copied to the local container's filesystem at
+  `/usr/local/bin/cephcsi`. This helps preventing any issues like intentional deletion of the
+  CSI-sidecars DaemonSet (it cleanups up the binary upon deletion) while the CSI-plugin DaemonSet is
+  still running.
+  There is obviously a performance downside to copying the binary and potential failures too.
```

Comment:
How big is the cephcsi binary? I'm imagining in the 30-50 MB range for a typical go binary. Hopefully perf won't be a real concern, especially if we could skip the copy if it's already the latest?